### PR TITLE
Cache publisher page.

### PR DESCRIPTION
**Pull request by isoos**
- #1518
- The auto-clearing cache may be a bit more than we actually need, but it has the benefit that the  potentially-mutating operations will clear the cache by default.

### CODE REVIEW

File: app/lib/publisher/backend.dart [hunk 1]
```diff
@@ -40,17 +41,6 @@ class PublisherBackend {
     return (await _db.lookup<Publisher>([pKey])).single;
   }
 
-  /// Checks whether the current authenticated user has admin role of the
-  /// publisher, and executes [fn] if it does.
-  /// Otherwise, it throws [AuthorizationException].
-  /// TODO: remove this and migrate to use withPublisherAdmin
-  Future<R> _withPublisherAdmin<R>(
```
Comment (jonasfj) [inline app/lib/publisher/backend.dart]: Hmm, can we have `withPublisherAdmin` call `withAuthenticatedUser`? Or does it need to be separate, it seems like two wrappers always used together...

We could also change these to: `final u = await ensureAuthenticatedUser();` and `await ensurePublisherAdmin(u);` just to reduce indentation... I guess the with-pattern is a bit safer though...

File: app/lib/publisher/backend.dart [hunk 2]
```diff
@@ -335,10 +354,14 @@ api.PublisherInfo _asPublisherInfo(Publisher p) =>
...
 ///
-/// Throws AuthenticationException if the user is provided.
+/// Throws AuthenticationException if the user is not provided.
 /// Throws AuthorizationException if the user is not an admin for the publisher.
 Future<R> withPublisherAdmin<R>(
-    String publisherId, String userId, FutureOr<R> fn(Publisher p)) async {
+  String publisherId,
+  String userId,
+  FutureOr<R> fn(Publisher p), {
+  bool clearCache = true,
```
Comment (jonasfj) [inline app/lib/publisher/backend.dart]: Let's not mix concerns... Can't we make a `_purgeCache(String publisherId)` on the backend instead?

Comment (isoos) [reply app/lib/publisher/backend.dart]: I've removed this from the PR, keeping only a separate `purgePublisherCache` method.

File: app/lib/publisher/backend.dart (hunk 1 above)
Comment (isoos) [reply app/lib/publisher/backend.dart]: Removed this from the PR, will do the other refactor separately.

**Pull request merged**